libyul/backends/evm/EVMDialect.cpp

```diff
@@ -142,6 +142,23 @@ map<YulString, BuiltinFunctionForEVM> createBuiltins(langutil::EVMVersion _evmVe
 			Expression const& arg = _call.arguments.front();
 			_assembly.appendLinkerSymbol(std::get<Literal>(arg).value.str());
 		}));
+
+		builtins.emplace(createFunction(
+			"memoryguard",
+			1,
+			1,
+			SideEffects{true, true, true, false, false},
```

Comment:
We should have some nicer notation for side-effects, but are these flags 100% correct?

Reply:
Yes, it's actually side-effect free. I can actually change it to ``SideEffects{}``, that's the same.